### PR TITLE
Add support for join org page

### PR DESCRIPTION
**Pull request by bduffany**
- Add `JoinGroup` RPC
- Update CSS
- Add illustration to `/static`
- Bonus (unrelated) bugfixes for edit/create pages

### CODE REVIEW

**Comment by bduffany** [comment]
Doesn't seem like GitHub will let me change this to "draft" so I can prevent submission until we add proper auth per the TODO in the comments, so will close this PR until we get proper auth protecting the `AddUserToGroup` call.

File: proto/grp.proto [hunk 1]
```diff
@@ -28,6 +28,14 @@ message Group {
   string url_identifier = 5;
 }
 
+message JoinGroupRequest {
```
Comment (tylerwilliams) [inline proto/grp.proto:31]: We've been lax about it (my fault! sorry) but all Request/Response pairs in here need to have:

context.RequestContext request_context = 1;
in the request and 
context.ResponseContext response_context = 1;
in the response.

Could you do that here or in a followup?

File: server/buildbuddy_server/buildbuddy_server.go [hunk 2]
```diff
@@ -171,14 +172,46 @@ func (s *BuildBuddyServer) CreateGroup(ctx context.Context, req *grpb.CreateGrou
...
+		// Otherwise submit a request to join the group.
+		if err := userDB.RequestToJoinGroup(ctx, user.UserID, req.GetId()); err != nil {
+			return nil, err
+		}
+	}
+
+	return &grpb.JoinGroupResponse{}, nil
+}
+
 func GetEmailDomain(email string) string {
```
Comment (tylerwilliams) [inline server/buildbuddy_server/buildbuddy_server.go:215]: can this be un-exported?

File: server/buildbuddy_server/buildbuddy_server.go [hunk 3]
```diff
@@ -171,14 +172,46 @@ func (s *BuildBuddyServer) CreateGroup(ctx context.Context, req *grpb.CreateGrou
...
+	}
+	group := &tables.Group{
+		GroupID: req.GetId(),
+	}
+	if err := userDB.FillGroup(ctx, group); err != nil {
+		return nil, err
+	}
+	// If the user's email matches the group's owned domain, they can be added
+	// as a member immediately.
+	if group.OwnedDomain != "" && GetEmailDomain(user.Email) == group.OwnedDomain {
```
Comment (tylerwilliams) [inline server/buildbuddy_server/buildbuddy_server.go:201]: might be slightly more readable if you swap the order of these two: GetEmailDomain(user.Email) == group.OwnedDomain

then you have if x != nil && x.y == z {}

**Review by tylerwilliams — CHANGES_REQUESTED** [review]

File: server/buildbuddy_server/buildbuddy_server.go (hunk 3 above)
Comment (bduffany) [reply server/buildbuddy_server/buildbuddy_server.go]: Done

File: proto/grp.proto (hunk 1 above)
Comment (bduffany) [reply proto/grp.proto:31]: Sure, let me do that in a follow up

File: server/buildbuddy_server/buildbuddy_server.go (hunk 2 above)
Comment (bduffany) [reply server/buildbuddy_server/buildbuddy_server.go]: Done

**Review by tylerwilliams — APPROVED** [review]
LGTM

**Pull request merged**